src/main/java/org/fenixedu/academic/ui/spring/AcademicBaseController.java

```diff
@@ -0,0 +1,39 @@
+package org.fenixedu.academic.ui.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+
+public class AcademicBaseController {
+    //The HTTP Request that can be used internally in the controller
```

Comment:
http://docs.spring.io/spring/docs/current/spring-framework-reference/html/mvc.html

Lookup the sections about form error handling.